File: Data Structures & Algorithms/two-integer-sum/submission-4.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        prevHashMap = {}
        for i in range(len(nums)):
           if( (target-nums[i]) in prevHashMap):
              return [prevHashMap[target-nums[i]], i]
           prevHashMap[nums[i]] = i

File: Data Structures & Algorithms/two-integer-sum-ii/submission-4.js
class Solution {
    /**
     * @param {number[]} numbers
     * @param {number} target
     * @return {number[]}
     */
    twoSum(numbers, target) {
        
        for(let i=0; i<numbers.length-1; ++i){
            let high=numbers.length-1;
            let low=i+1;
            while(low <= high){
                let mid = low + Math.floor((high-low)/2);
                if(numbers[mid] == target-numbers[i]){
                   return [i+1, mid+1];
                }
                else if(numbers[mid] < target-numbers[i]){
                   low = mid+1;
                }
                else{
                   high = mid-1;
                }
            }
        }

        //no solution
        return [-1,-1];
    }
}
